tests/run_all_tests.sh

```
@@ -0,0 +1,387 @@
+    RUN_NESTEST=true
+fi
+
+# Results tracking
+TOTAL_TESTS=0
+PASSED_TESTS=0
+FAILED_TESTS=0
+RESULTS_FILE="test_results.txt"
+JSON_FILE="test_results.json"
+
+# Clear previous results
+> "$RESULTS_FILE"
```

Shellcheck (0.11.0)

[warning] This redirection doesn't have a command. Move to its command (or use 'true' as no-op).

(SC2188)